core/domain/suggestion_registry.py

```diff
@@ -635,6 +635,209 @@ def pre_update_validate(self, change):
+
+    @property
+    def is_handled(self):
+        """Returns true if the voiceover application has either been accepted or
+        rejected.
+
+        Returns:
+            bool. Whether the voiceover application has been handled or not.
+        """
+        return self.status != suggestion_models.STATUS_IN_REVIEW
+
+
+class ExplorationVoiceoverApplication(BaseVoiceoverApplication):
+    """Domain object for a voiceover application for exploration."""
+
+    def __init__( # pylint: disable=super-init-not-called
+            self, voiceover_application_id, target_id, status, author_id,
+            final_reviewer_id, language_code, filename, content,
+            rejection_message):
+        """Initializes a ExplorationVoiceoverApplication domain object.
+
+        Args:
+            voiceover_application_id: str. The ID of the voiceover application.
+            target_id: str. The ID of the target entity.
+            status: str. The status of the voiceover application.
+            author_id: str. The ID of the user who submitted the voiceover
+                application.
+            final_reviewer_id: str|None. The ID of the reviewer who has
+                accepted/rejected the voiceover application.
+            language_code: str. The language code for the voiceover application.
+            filename: str. The filename of the voiceover audio.
+            content: str. The html content which is voiceover in the
+                application.
+            rejection_message: str. The plain text message submitted by the
+                reviewer while rejecting the application.
+        """
+        self.voiceover_application_id = voiceover_application_id
+        self.target_type = suggestion_models.TARGET_TYPE_EXPLORATION
+        self.target_id = target_id
+        self.status = status
+        self.author_id = author_id
+        self.final_reviewer_id = final_reviewer_id
+        self.language_code = language_code
+        self.filename = filename
+        self.content = content
+        self.rejection_message = rejection_message
+
+    def accept(self, reviewer_id):
+        """Accepts the voiceover application and updates the final_reviewer_id.
+
+        Args:
+            reviewer_id: str. The user ID of the reviewer.
+        """
+        self.final_reviewer_id = reviewer_id
+        self.status = suggestion_models.STATUS_ACCEPTED
+        self.validate()
+
+    def reject(self, reviewer_id, rejection_message):
+        """Rejects the voiceover application, updates the final_reviewer_id and
+        adds rejection message.
+
+        Args:
+            reviewer_id: str. The user ID of the reviewer.
+            rejection_message: str. The rejection message submitted by the
+                reviewer.
+        """
+        self.status = suggestion_models.STATUS_REJECTED
+        self.final_reviewer_id = reviewer_id
+        self.rejection_message = rejection_message
+        self.validate()
```

Comment:
ditto, validate before rejecting. (maybe validate after also, to make sure it is still valid).

Reply:
Do we still need this considering this [thread](https://github.com/oppia/oppia/pull/7843#discussion_r340296912)?